### PR TITLE
EnvFlags OK

### PR DESCRIPTION
**Pull request by hensldm**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Copy paste of OOT's code_8006C3A0. I think I would prefer the system to be `EnvFlags` not `Flags` since those are the only flags this is touching, however decided to just use what OOT already had. Would love other's opinions though.

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]
`EnvFlags` sounds good to me. Would be nice to hear more feedback about it tho

**Review by Kenix3 — COMMENTED** [review]
I agree with the EnvFlags system rename. Also, we should rename the code file.

**Comment by hensldm** [comment]
Okay, renamed the system to EnvFlags and the file to z_env_flags.c.

**Review by Kenix3 — APPROVED** [review]
I wonder how hard it would be to document a flag enum. Itd require reading the asm most likely.

File: src/overlays/actors/ovl_Door_Ana/z_door_ana.c [hunk 1]
```diff
@@ -92,7 +92,7 @@ void DoorAna_WaitClosed(DoorAna* this, GlobalContext* globalCtx) {
 
     if (grottoType == DOORANA_TYPE_UNK) {
         // in OOT decomp its marked as open with storms, but does not seem to open with storms in MM
-        if ((this->actor.xyzDistToPlayerSq < 40000.0f) && (func_800F13E8(globalCtx, 5))) {
+        if ((this->actor.xyzDistToPlayerSq < 40000.0f) && (EnvFlags_Get(globalCtx, 5))) {
```
Comment (isghj5) [inline src/overlays/actors/ovl_Door_Ana/z_door_ana.c:95]: oooo interesting, so that door_ana variant code is dependent on these flags?

**Review by isghj5 — APPROVED** [review]

**Pull request merged**